components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIMgtAdminServiceImpl.java

```diff
@@ -217,4 +217,29 @@ public String getAPIGatewayServiceConfig(String apiId) throws APIConfigRetrieval
         }
     }
 
+    @Override public List<API> getAPIsByStatus(List<String> gatewayLabels, String status)
+            throws APIManagementException {
+        List<API> apiList;
+        try {
+            apiList = apiDAO.getAPIsByStatus(gatewayLabels, status);
```

Comment:
null check is not done for gatewayLabels. there are couple of places in apiDAO.getAPIsByStatus() that could throw NPE if gatewayLabels == null